Rubberduck.VBEditor.VBA/SafeComWrappers/VB/VBProjects.cs

```diff
@@ -96,23 +96,25 @@ void VB._dispVBProjectsEvents.ItemRemoved([MarshalAs(UnmanagedType.Interface), I
         void VB._dispVBProjectsEvents.ItemRenamed([MarshalAs(UnmanagedType.Interface), In] VB.VBProject VBProject,
             [MarshalAs(UnmanagedType.BStr), In] string OldName)
         {
-            var project = new VBProject(VBProject);
-
-            if (!IsInDesignMode() || VBProject.Protection == VB.vbext_ProjectProtection.vbext_pp_locked)
+            using (var project = new VBProject(VBProject))
             {
-                project.Dispose();
-                return;
-            }
+                if (!IsInDesignMode() || VBProject.Protection == VB.vbext_ProjectProtection.vbext_pp_locked)
+                {
+                    project.Dispose();
+                    return;
+                }
 
-            var projectId = project.ProjectId;
+                var projectId = project.ProjectId;
 
-            var handler = ProjectRenamed;
-            if (handler == null || projectId == null)
-            {
-                project.Dispose();
-                return;
+                var handler = ProjectRenamed;
+                if (handler == null || projectId == null)
+                {
+                    project.Dispose();
+                    return;
```

Comment:
See above - the `return` should already be calling `Dispose` as the block exits.